### PR TITLE
test: read client

### PR DESCRIPTION
**Pull request by dcaf-mocha**

### CODE REVIEW

File: sdk/pre-authorized-debit-v1/src/read/implementation/index.ts [hunk 1]
```diff
@@ -65,7 +66,9 @@ export class PreAuthorizedDebitReadClientImpl
   }
 
   public async fetchIdl(): Promise<Idl> {
-    const idl = await Program.fetchIdl(this.programId);
+    const idl = await Program.fetchIdl(this.programId, {
+      connection: this.connection,
+    });
```
Comment (dcaf-mocha) [inline sdk/pre-authorized-debit-v1/src/read/implementation/index.ts:71]: This was a bug

File: .github/workflows/program-tests.yml [hunk 2]
```diff
@@ -67,9 +67,10 @@ jobs:
       matrix:
         command:
           [
-            "anchor test pre-authorized-debit-v1 .ts",
-            "anchor test pre-authorized-debit-v1 .2.ts",
-            "anchor test pre-authorized-debit-v1/debit .ts",
+            "anchor test program-tests/pre-authorized-debit-v1 .ts",
+            "anchor test program-tests/pre-authorized-debit-v1 .2.ts",
+            "anchor test program-tests/pre-authorized-debit-v1/debit .ts",
+            "anchor test psdk/pre-authorized-debit-v1/tests/**",
```
Comment (dcaf-matcha) [inline .github/workflows/program-tests.yml:73]:
```suggestion
            "anchor test sdk/pre-authorized-debit-v1/tests/**",
```

File: sdk/pre-authorized-debit-v1/src/read/implementation/index.ts (hunk 1 above)
Comment (dcaf-matcha) [reply sdk/pre-authorized-debit-v1/src/read/implementation/index.ts]: I renamed the function, can you remove this one?

**Review by dcaf-matcha — COMMENTED** [review]
LGTM except for couple changes

File: sdk/pre-authorized-debit-v1/src/read/implementation/index.ts (hunk 1 above)
Comment (dcaf-mocha) [reply sdk/pre-authorized-debit-v1/src/read/implementation/index.ts]: already done

**Pull request merged**